zerver/views/user_groups.py

```diff
@@ -115,6 +119,54 @@ def update_user_group_backend(
     return json_success(request, data)
 
 
+def send_messages_for_user_groups(
+    acting_user: UserProfile,
+    recipient_users: List[UserProfile],
+    user_group_name: str,
+    *,
+    send_subscription_message: bool,
+    send_unsubscription_message: bool,
+) -> None:
+    realm = acting_user.realm
+    mention_backend = MentionBackend(realm.id)
+
+    notifications = []
+    sender = get_system_bot(settings.NOTIFICATION_BOT, realm.id)
+    for recipient_user in recipient_users:
+        if recipient_user.email == acting_user.email:
+            # Don't send notification message if you subscribed/unsubscribed yourself.
+            continue
+        if recipient_user.is_bot:
+            # Don't send notification message to bots.
+            continue
```

Comment:
What happens if someone tries to edit the group membership of a deactivated user? We certainly don't want to send these notifications to deactivated users. I recommend including a deactivated user in your test setup; it'll be a case we want the tests to enforce since it's the kind of thing that tends to get messed up when refactoring.

Reply:
Earlier I missed the case of deactivated user but now I have done the changes and message will not be sent to deactivated user. Also added a test for this case.